articles/creating-a-hangman-game-with-vanilla-javascript/index.md

```diff
@@ -0,0 +1,238 @@
+```
+
+.
+##The Actual JavaScript
+First thing you need to do is to create an array containing all the words to be guessed inside the game. I used social media as my niche of words. You can use any one of your choice. Here's how it looks on your code.
+```javascript
+var socialMedia = [
+  "whatsapp",
+  "instagram",
+  "twitter",
+  "snapchat",
+  "tiktok",
+  "youtube",
+  "wechat",
+  "facebook",
+  "telegram",
+  "bbm",
+  "palmchat",
+  "gmail",
+  "2go",
+  "linkedin",
+  "reddit",
+  "hangouts",
+  "qq",
+  "twitch",
+  "douyin"
+]
+```
+
+These are words that are going to be randomized in the game.
+Next, you have to define the id's you created. Here it is in the code: By default,
+```javascript
+let answer = '';
+let maxWrong = 6;
+let mistakes = 0;
+let guessed = [];
+```
+
+The "answer" id contains the correct word to be guessed. The "maxWrong" id contains the maximum number of times you can guess wrong. The "mistakes" id contains the number of times you have guessed wrong so far, while the "guessed" id contains the letters you will guess.
+##The Game Functions
+###randomWord() function
+This function will help to randomly pick any of the words you placed in your social media array, which now makes the game more interesting as you do not know what word you are guessing next. Here is how the code looks like:
+function randomWord() {
+  answer = socialMedia[Math.floor(Math.random() * socialMedia.length)];
+}
+
+Then you call the function by using the code below:
+randomWord();
+
+###generateButtons() function
+If you notice, there are alphabet buttons in our HTML. Without them, how are we supposed to guess the words? Well, this function helps us generate those buttons. This is a better method of creating these buttons rather than creating 26 different button tags in the HTML file. It is a good representation of "don’t-repeat-yourself" coding. Here's the code below:
+```javascript
+function generateButtons() {
+  let buttonsHTML = 'abcdefghijklmnopqrstuvwxyz'.split('').map(letter =>
+    `
+      <button
+        class="btn btn-lg btn-primary m-2"
+        id='` + letter + `'
+        onClick="handleGuess('` + letter + `')"
+      >
+        ` + letter + `
+      </button>
+    `).join('');
+
+  document.getElementById('keyboard').innerHTML = buttonsHTML;
+}
+```
+
+
+In the code above, we defined all 26 letter buttons while styling them with Bootstrap. "letter" in the code represents each of the looped items in the array of alphabets that we just created in the variable. You then connect it to your HTML using the `document.getElementById('keyboard').innerHTML = buttonsHTML`. Remember, we defined the "keyboard" id in the HTML. Also, ` .join('')` was added to eliminate the commas between the alphabet buttons.
+
+Then you run it with the code below:
+```javascript
+generateButtons();
+```
+
+Here's what the game looks like at this point:
+![hangman 1](/engineering-education/creating-a-hangman-game-with-vanilla-javascript/hangman 1.jpg)
+
```

Comment:
Provide descriptive image names and alt text. Please rename your images. Replace the spaces in the filenames with hyphens (-).